crates/bevy_ecs/src/reflect.rs

```diff
@@ -265,7 +264,7 @@ pub struct ReflectResourceFns {
     /// Function pointer implementing [`ReflectResource::reflect()`].
     pub reflect: fn(&World) -> Option<&dyn Reflect>,
     /// Function pointer implementing [`ReflectResource::reflect_unchecked_mut()`].
```

Comment:
This fn pointer should have safety comment, especially since the field is public.